omnigibson/prims/joint_prim.py

```diff
@@ -224,11 +206,13 @@ def set_control_type(self, control_type, kp=None, kd=None):
             kp, kd = 0.0, 0.0
 
         # Set values
-        if self._dc:
-            for dof_handle, dof_property in zip(self._dof_handles, self._dof_properties):
-                dof_property.stiffness = kp
-                dof_property.damping = kd
-                self._dc.set_dof_properties(dof_handle, dof_property)
+        kps = np.full((1, self._n_dof), kp)
+        kds = np.full((1, self._n_dof), kd)
+        self._articulation_view.set_gains(kps=kps, kds=kds, joint_indices=self.dof_indices, save_to_usd=True)
+
+        # Assert no weirdness happening around the save_to_usd business.
+        kps, kds = self._articulation_view.get_gains(joint_indices=self.dof_indices)
+        assert np.allclose(kps, kp) and np.allclose(kds, kd), "Something went wrong with setting the gains!"
```

Comment:
there is a problem here that the getter/setter for gains has the unique save_to_usd feature that makes the gains update on the USD instead of in fabric. it is unclear if this will always update fabric too, hence the check. however, that feature is not aligned with all the other setters so I've gone and disabled it now so no need for the check